frontend/src/components/KeyMeasureDiagnostic/ExpeModals/ExpeVegetarian.vue

```diff
@@ -0,0 +1,788 @@
+            >
+              <v-tab v-for="item in tabs" :key="item.key">
+                {{ item.label }}
+              </v-tab>
+            </v-tabs>
+            <v-tabs-items v-model="tab">
+              <v-tab-item v-for="item in tabs" :key="item.label">
+                <v-card flat>
+                  <v-card-text>
+                    <!-- Vegetarian menu rate -->
+                    <label
+                      class="body-2 grey--text text--darken-3 font-weight-medium"
+                      :for="`vegetarian-menu-percentage-${item.value}`"
+                    >
+                      Pourcentage de menus végétariens servis par rapport aux autres menus
+                    </label>
+                    <v-text-field
+                      validate-on-blur
+                      hide-details="auto"
+                      :rules="[validators.nonNegativeOrEmpty]"
+                      solo
+                      dense
+                      v-model.number="expe[`vegetarianMenuPercentage${item.value}`]"
+                      append-icon="mdi-percent"
+                      style="max-width: 150px"
+                      class="mt-2 mb-4 body-2"
+                      :id="`vegetarian-menu-percentage-${item.value}`"
+                    ></v-text-field>
+
+                    <!-- Menu composition -->
+                    <label class="body-2 grey--text text--darken-3 font-weight-medium">
+                      Dans votre plan alimentaire de 20 repas successifs, les options végétariennes servies dans votre
+                      établissement sont des plats à base de :
+                    </label>
+                    <div class="mt-4 mb-8" style="border-bottom: solid 1px #EEE;">
+                      <div
+                        class="d-flex py-2"
+                        style="border-top: solid 1px #EEE;"
+                        v-for="category in categories"
+                        :key="category.htmlId"
+                      >
+                        <label
+                          class="body-2 grey--text text--darken-3 pr-2"
+                          :style="$vuetify.breakpoint.mdAndUp ? 'width: 35%;' : 'width: 70%;'"
+                          :for="`${category.htmlId}-${item.value}`"
+                        >
+                          {{ category.label }}
+                        </label>
+                        <div class="d-flex">
+                          <v-text-field
+                            validate-on-blur
+                            hide-details="auto"
+                            solo
+                            dense
+                            v-model.number="expe[`${category.fieldName}${item.value}`]"
+                            :rules="[
+                              validators.nonNegativeOrEmpty,
+                              validators.lteSumValue(
+                                self[`compositionValidationFields${item.value}`],
+                                20,
```

Comment:
This is a bit buggy - the validation isn't reset if the error is flagged on one field but then the value of another field is changed instead.
Maybe only add this validation to the last field, make it validate on submit instead of blur ? Still not ideal but might be a way of stopping it from blocking (in the case above I have to click submit twice for the out of date validation error to be ignored)

Reply:
+1 for validation on submit, with a revalidation of all fields on the blur of one of them

Reply:
https://github.com/betagouv/ma-cantine/pull/1237/commits/9a2c6fcbcf963326f0791776008c6093ab5d9a2f